### PR TITLE
Implement new quest functions, tests, and data structures

### PR DESCRIPTION
**Pull request by ddelgado60**
This branch is an extension off of quests/implement-trees to preserve work done in that branch, and also implement new tests, functions, and data structures for passive/active quests, stats requirements, and reward types. There is an existing pull request for implement new quest structs, but it seems more strategic to branch off of quests/implement-trees, since work done there was necessary for work done in this branch.

### CODE REVIEW

File: tests/quests/test_quests_state.c [hunk 1]
```diff
@@ -27,71 +29,155 @@ class_t* generate_test_class()
...
+
+    return stat_req;
+}
+
+/* Tests init function for quest struct */
+Test(quest, init)
+{   
+    quest_t *q = malloc(sizeof(quest_t));
+    reward_t *rewards = create_sample_rewards();
+    stat_req_t *stat_req = create_sample_stat_req();
```
Comment (gabigarc03) [inline tests/quests/test_quests_state.c:169]: Do these two functions (`create_sample_rewards()` and `create_sample_stat_req()`) necessarily need to be functions? Alternatively, there might be a benefit to allowing them to take input to make them more flexible and usable for more extensive testing.

File: tests/quests/test_quests_state.c [hunk 2]
```diff
@@ -149,12 +253,97 @@ Test(achievement, free)
...
+active_mission_t *make_example_a_mission()
+{
+    class_t* class = generate_test_class();
+    char *npc_meet_id = "meet_npc";
+    char *npc_kill_id = "kill_npc";
+
+    item_t *item_to_get = item_new("test_item", "item for testing",
+    "test item for item_new()");
+    npc_t *mission_meet_npc = npc_new(npc_meet_id ,"npc1", "npc to meet",
+                                100, class);
```
Comment (gabigarc03) [inline tests/quests/test_quests_state.c:265]: Make sure you're initializing these npcs properly

File: tests/quests/test_quests_state.c [hunk 3]
```diff
@@ -149,12 +253,97 @@ Test(achievement, free)
...
+
+    npc_t *mission_meet_kill = npc_new(npc_kill_id ,"npc2", "npc to kill", 
+                                       100, class);
```
Comment (gabigarc03) [inline tests/quests/test_quests_state.c:268]: Make sure you're initializing these npcs properly

File: tests/quests/test_quests_state.c [hunk 4]
```diff
@@ -166,42 +355,64 @@ Test(quest, free)
...
-    achievement_t *achievement = achievement_new(mission, "mission1");
+    active_mission_t *a_mission = active_mission_new(item_to_get, NULL, NULL, NULL);
+    mission_t *mission;
+    mission->a_mission = a_mission;
+    mission->p_mission = NULL;
 
-    int res = add_achievement_to_quest(quest, achievement, "NULL");
+	achievement_t* achievement_to_free = achievement_new(mission, id);
+
+    int res = add_achievement_to_quest(quest, achievement_to_free, "NULL");
```
Comment (gabigarc03) [inline tests/quests/test_quests_state.c:373]: Why is this called `achievement_to_free` when the function being tested is `add_achievement_to_quest`?

File: tests/quests/test_quests_state.c [hunk 5]
```diff
@@ -228,19 +440,21 @@ Test(quest, fail_quest)
...
+    active_mission_t *a_mission = make_example_a_mission();
+
+
+    char *id = "test mission";
+
+    mission_t *mission;
+    mission->a_mission = a_mission;
+    mission->p_mission = NULL;
+
+	achievement_t* achievement_to_free = achievement_new(mission, id);
```
Comment (gabigarc03) [inline tests/quests/test_quests_state.c:457]: Why is this called `achievement_to_free` when it's in a function to test `complete_achievement`?

**Review by gabigarc03 — CHANGES_REQUESTED** [review]
Overall - this looks good! Just left a bunch of minor comments because I'm pedantic.

**Review by gabigarc03 — APPROVED** [review]

**Comment by gabigarc03** [comment]
These changes look good!

**Review by lilyehsani — APPROVED** [review]
LGTM! Approved.

**Comment by lilyehsani** [comment]
Issue Score: ✔️

Comments:
Great work overall! I have a few comments:
- Like issues, PRs require closing comments. When an PR is done, you should include a final comment summarizing what was merged to `dev` and what work remains to be done, if any.
- Make sure to add anyone who is working on this PR as an Assignee
- Make sure to assign new PRs to the sprint they are worked on (by selecting the appropriate sprint under "Milestone")

**Pull request merged**